### PR TITLE
Fix multiple history issues

### PR DESCRIPTION
**Pull request by bena-nasa**
Fixes: 
#1607
#1285
#1376
#1374
#626 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [X] I have tested this change with a run of GEOSgcm (if non-trivial)
- [X] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [X] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Review by atrayano — COMMENTED** [review]
@bena-nasa Are we sure the new alias checking works well with the field splitting. My concern is the timing of the check - before we do the splitting, the alias field might "look the same' until is expanded during the splitting

**Comment by bena-nasa** [comment]
> @bena-nasa Are we sure the new alias checking works well with the field splitting. My concern is the timing of the check - before we do the splitting, the alias field might "look the same' until is expanded during the splitting

I'll check that I haven't messed up the field splitting

File: gridcomps/History/MAPL_HistoryGridComp.F90 [hunk 1]
```diff
@@ -3323,6 +3327,22 @@ subroutine parse_fields(cfg, label, field_set, items, rc)
           if(present(items)) call items%push_back(item)
        enddo
        field_set%nfields = m
+!      check for duplicates
+       do i=1,field_set%nfields-1
+          do j=i+1,field_set%nfields
+             match_short_name = trim(field_set%fields(1,i)) == trim(field_set%fields(1,j))
```
Comment (mathomp4) [inline gridcomps/History/MAPL_HistoryGridComp.F90:3333]: I'm not sure the `trim()` are needed here. @tclune would know for sure, but I think when you compare strings, like `x == y` then it just goes character by character to the right and as long as all characters in `x` are the same as `y`, it's equal. 

And I think if you have unequal strings, then the shorter is padded with spaces?

**Comment by bena-nasa** [comment]
> @bena-nasa Are we sure the new alias checking works well with the field splitting. My concern is the timing of the check - before we do the splitting, the alias field might "look the same' until is expanded during the splitting

A split collection was zero-diff when I checked a split collection vs the control.

**Review by atrayano — APPROVED** [review]

File: griddedio/GriddedIO.F90 [hunk 2]
```diff
@@ -254,6 +257,35 @@ subroutine set_default_chunking(this,rc)
...
+        integer, intent(in) :: lev_size
+        integer, optional, intent(out) :: rc
+
+        integer ::  global_dim(3)
+        integer :: status
+
+        call MAPL_GridGet(this%output_grid,globalCellCountPerDim=global_dim,rc=status)
+        _VERIFY(status)
+        if (global_dim(1)*6 == global_dim(2)) then
+           _ASSERT(this%chunking(1) <= global_dim(1), "Bad chunk size")
```
Comment (mathomp4) [inline griddedio/GriddedIO.F90:271]: Should we have more "explicit" asserts here? Like "chunk for X dim, this%chunking(1), must be less than or equal to global_dim(1)"? Not sure.

File: gridcomps/History/MAPL_HistoryGridComp.F90 (hunk 1 above)
Comment (bena-nasa) [reply gridcomps/History/MAPL_HistoryGridComp.F90]: Yep, appear to not need the trims

File: griddedio/GriddedIO.F90 (hunk 2 above)
Comment (bena-nasa) [reply griddedio/GriddedIO.F90]: Added informative messages

**Review by mathomp4 — APPROVED** [review]

**Pull request merged**